### PR TITLE
Rename `DataEntity` to `PackageType`

### PR DESCRIPTION
**Pull request by mnonnenmacher**
`DataEntity` is a very generic name, replace it with a more specific
name that better describes the purpose of the enum. While at it, also
improve the documentation of the enum and rename the command line
parameters accordingly.

### CODE REVIEW

**Review by fviernau — DISMISSED** [review]
Out of scope: Would it be a good idea to turn the plural enum elements into singular ones in a separate PR ?

**Comment by sschuberth** [comment]
Note that *detekt* analysis failed.

**Comment by mnonnenmacher** [comment]
> Out of scope: Would it be a good idea to turn the plural enum elements into singular ones in a separate PR ?

I have added a commit for that.

File: cli/src/main/kotlin/commands/DownloaderCommand.kt [hunk 1]
```diff
@@ -213,11 +213,11 @@ class DownloaderCommand : CliktCommand(name = "download", help = "Fetch source c
         }
 
         val packages = mutableListOf<Package>().apply {
-            if (PackageType.PROJECTS in packageTypes) {
+            if (PackageType.PROJECT in packageTypes) {
```
Comment (sschuberth) [inline cli/src/main/kotlin/commands/DownloaderCommand.kt:216]: Nit: I think this commit should then also adjust the `Downloading ${packageTypes.joinToString(" and ")} from ORT` code from the previous commit by adding a plural "s" after `}`.

File: cli/src/main/kotlin/commands/DownloaderCommand.kt [hunk 2]
```diff
@@ -196,7 +196,7 @@ class DownloaderCommand : CliktCommand(name = "download", help = "Fetch source c
 
     private fun downloadFromOrtResult(ortFile: File, failureMessages: MutableList<String>) {
         println(
-            "Downloading ${packageTypes.joinToString(" and ")} from ORT result file at " +
+            "Downloading ${packageTypes.joinToString(" and ") { "${it}s" }} from ORT result file at " +
```
Comment (fviernau) [inline cli/src/main/kotlin/commands/DownloaderCommand.kt:199]: The curly braces around `it` are unnecessary.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: cli/src/main/kotlin/commands/DownloaderCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/DownloaderCommand.kt:199]: I don't think so, as you need to separate from the "s".

**Review by sschuberth — APPROVED** [review]

File: cli/src/main/kotlin/commands/DownloaderCommand.kt (hunk 2 above)
Comment (fviernau) [reply cli/src/main/kotlin/commands/DownloaderCommand.kt:199]: OMG, you're right ^^

**Review by fviernau — APPROVED** [review]

**Pull request merged**